experimental/trusted_shuffler/trusted_shuffler/src/tests.rs

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::TrustedShuffler;
+use futures::future::join_all;
+use std::{sync::Arc, time::Duration};
+use tokio::time::timeout;
+
+const TEST_ANONYMITY_VALUE: usize = 10;
+const TEST_TIMEOUT_MILLISECONDS: u64 = 100;
+
+async fn send_request(request: Vec<u8>) -> Vec<u8> {
+    request
+}
+
+#[tokio::test]
+async fn trusted_shuffler_test() {
+    let trusted_shuffler = Arc::new(TrustedShuffler::new(2, send_request));
+
+    let request = "Test request".to_string().as_bytes().to_vec();
+    let background_request = "Test request background".to_string().as_bytes().to_vec();
+
+    let trusted_shuffler_clone = trusted_shuffler.clone();
+    let background_request_clone = background_request.clone();
+    let background_result = tokio::spawn(async move {
+        trusted_shuffler_clone
+            .invoke(background_request_clone)
+            .await
+    });
+
+    let response = trusted_shuffler.invoke(request.clone()).await;
+    assert!(response.is_ok());
+    assert_eq!(request, response.unwrap());
```

Comment:
Here we assume that our "test backend" echoes the the request, yes? Maybe we want to comment this here. Or maybe even remove this assertion. We just want to know we got a response, no?